### ISSUE TITLE
--path default not working

### ISSUE THREAD

**Issue by iemejia** [coursera-dl/coursera-dl#390]
Hi,

This is a really minor issue, I was running the script without putting the --path argument and I notice it didn't asume the current path as the default

```
python coursera-dl -n cloudapplications-001
coursera-dl: error: too few arguments
```

I suppose this can be easily patched changing the default to '.' in line:
https://github.com/coursera-dl/coursera-dl/blob/7ce7d56b04c560c2bc0a41878d6058308305cd61/coursera/coursera_dl.py#L657

Is there any reason not to do it, or this is just a bug ?

**Comment by mxamin** [comment]
I don't think it's a bug, you entered the command wrong. try:
`python coursera-dl -n -- cloudapplications-001`

**Comment by iemejia** [comment]
I suppose '--' is the convention for end of arguments, is it ? But a casual user of the script will expect it to work without the arguments since coursa-dl --help says nothing about '--', but it says:
```
--path PATH           path to where to save the file. (Default: current directory)
```

**Comment by mxamin** [comment]
@iemejia , Yes, that's right `--` means the end of argument, and it's used because `-n` accept zero or one argument, and without using double dash, it will consider the next word (which is the course name) as it's own argument.
Now, about `--path` option. Based on what I understood from your comment, you expect it work without `PATH` argument, because of *Default* work in option description. Actually it means if you don't use this option **at all**, the current directory will be chosen as destination.
I hope it will help you with your problem.

P.S: empty directory equals `./` directory in *Python*.

**Comment by iemejia** [comment]
Hi, thanks, well this is not a problem as I said, but just in my opinion a misconfigured default, or missing documentation, notice that the current default for the script is '' and not './' as you mentioned.
Well anyhow, feel free to fix (If you consider I can do the PR), or just close the issue if you consider it is not worth.

Best,
-Ismael

**Comment by mxamin** [comment]
Feel free to add PR, clearly it makes the code more reliable.
thanks.

**Comment by rbrito** [comment]
@iemejia, there is, indeed, some ambiguity in the way that we would like argparse to work here, as @mxamin well exposed.

Of course, since it can't read our minds, I am thinking of deprecating the current behavior, leave the option `-n`/`--netrc` as boolean and add another `--netrc-path` that would allow the users to specify where the user put his/her `.netrc` file.

As soon as I added that `-n` option, I was faced myself with the need to use that dreaded `--`, which I promptly documented. (read: "I use the `--` myself").

For the future, migrating away from `.netrc` files is a better/more secure option, but I'll leave that as the subject for the future.

For now, what do you guys think of the `--netrc`/`--netrc-path` pair of options?


Regards,

Rogério.

**Comment by iemejia** [comment]
Hi @rbrito, you are discussing about a different issue, I don't have problems with the path of .netrc which works well for me (but I have seen people in particular who use windows have issues with), what I just discussed is about the --path command option to choose where to download the course material which has a bad (or at least confusing) default.

Talking about the issue you mention, I understand the importance of having the path option, however I will keep the current behavior (which I suppose looks for the file based on the $HOME var), and fix a bit the documentation to tell the users where to put the file, or how to overwrite the path.

**Comment by rbrito** [comment]
@iemejia, oh, darn. I was discussing, indeed, another issue. :) See? I'm getting senile. :)

Regarding the --path option, I always thought that the current behavior was OK. If not, please go ahead and propose a patch to the documentation or even to the program.

I have taken care to not declare the program "stable", since I may want to revisit the semantic of the options.  With a "grace period" and scandalous output messages, I have no problems in changing the behavior of the options, if you see the need to.
